esp/services/ws_fs/ws_fsService.cpp

```diff
@@ -1833,13 +1833,14 @@ bool CFileSprayEx::onGetDFUExceptions(IEspContext &context, IEspGetDFUExceptions
 
 static StringBuffer &getDropZoneHost(const char *planeName, IPropertyTree *plane, StringBuffer &host)
 {
-    if (plane->hasProp("hosts"))
+    if (plane->hasProp("@hostGroup"))
```

Comment:
I think we support both hosts and @hostGroup.  On baremetal I think @hostGroup is expanded into hosts.  Probably should cope with both.

Reply:
>On baremetal I think @hostGroup is expanded into hosts. 

it wasn't which is why it caused the problem to be highlighted.

I think it's the other way around, in bare-metal afaics, it's always a reference to a @hostGroup at the moment, as per constructed from env. via generateHosts and normalizeHostGroups.

In cloud, the schema allows for both.

So I think in bare-metal (at least at moment) it's always a @hostGroup ref., and in cloud it could be either.

But yes, that implies this code must support both too.
I'll change it.

Reply:
@ghalliday - please see reworked code/added helpers.